### PR TITLE
Ignite 8776:Eviction policy MBeans are never registered if evictionPolicyFactory is used

### PR DESCRIPTION
**Pull request by kcmvp**
Please help to review it. Thanks

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/GridMBeansTest.java [hunk 1]
```diff
@@ -60,6 +100,22 @@ public void testKernalBeans() throws Exception {
         checkBean("Kernal", "ClusterMetricsMXBeanImpl", "TotalServerNodes", 1);
     }
 
+    /** Check that eviction bean is available */
+    public void testEvictionPolicyBeans() throws Exception{
```
Comment (slukyano) [inline modules/core/src/test/java/org/apache/ignite/internal/GridMBeansTest.java]: Let's move this test case to a separate test (say, a new test called org.apache.ignite.internal.processors.cache.GridEvictionPolicyMBeansTest).

**Comment by kcmvp** [comment]
@slukyano I did the changes according your comments. please review it. Thank you.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        cache1.setName("cache1");
+        cache1.setOnheapCacheEnabled(true);
+        cache1.setEvictionPolicyFactory(plc);
+
+        NearCacheConfiguration ncf = new NearCacheConfiguration<>();
+        LruEvictionPolicy lep = new LruEvictionPolicy();
+        lep.setBatchSize(10);
+        lep.setMaxMemorySize(500);
+        lep.setMaxSize(40);
+        ncf.setNearEvictionPolicy(lep);
```
Comment (slukyano) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java]: Let's use `setNearEvictionPolicyFactory` here.

I think we need 4 cases here: `setEvictionPolicyFactory`, `setNearEvictionPolicyFactory`, `setEvictionPolicy` and `setNearEvictionPolicy`.

You can use factories in `cache1` and plain policies in `cache2`.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    /** Create test and auto-start the grid */
+    public GridEvictionPolicyMBeansTest() {
+        super(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This implementation registers adds custom executors to the configuration.
```
Comment (slukyano) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java]: "registers adds custom executors" -> "adds eviction policies".

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEvictionManager.java [hunk 4]
```diff
@@ -311,4 +314,126 @@ private void notifyPolicy(GridCacheEntryEx e) {
...
     }
+
+    /**
+     * @param cfg Cache configuration.
+     * @param rsrc Resource.
+     * @param near Near flag.
+     * @throws IgniteCheckedException If failed.
+     */
+    private void prepare(CacheConfiguration cfg, @Nullable Object rsrc, boolean near) throws IgniteCheckedException {
+
```
Comment (slukyano) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEvictionManager.java]: Unnecessary empty line

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEvictionManager.java [hunk 5]
```diff
@@ -71,9 +73,10 @@ else if (cfg.getEvictionPolicyFactory() != null)
             plc = cfg.getEvictionPolicy();
 
         plcEnabled = plc != null;
+        if (plcEnabled)
+            prepare(cfg,plc,cctx.isNear());
```
Comment (slukyano) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEvictionManager.java]: Need spaces after commas.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        lep = new LruEvictionPolicy();
+        lep.setBatchSize(10);
+        lep.setMaxMemorySize(125);
+        lep.setMaxSize(30);
+
+        cache2.setEvictionPolicy(lep);
+        cache2.setOnheapCacheEnabled(true);
+        cache2.setName("cache2");
+
+        cfg.setCacheConfiguration(cache1,cache2);
```
Comment (slukyano) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java]: Need space after comma.

**Comment by kcmvp** [comment]
@slukyano I did the fix according to your comments. Help to review, please!

**Pull request closed without merging**